download/download_test.go

```diff
@@ -253,50 +253,85 @@ func TestEtagCachingLifecycle(t *testing.T) {
 		t.Fatalf("Expected empty downloader ETag but got %v", fixture.d.etag)
 	}
 
-	// simulate successful bundle activation and check updated etag on the downloader
+	// simulate downloader error on first bundle download
+	fixture.server.expCode = 500
 	fixture.server.expEtag = "some etag value"
 	_, err := fixture.d.oneShot(ctx)
+	if err == nil {
+		t.Fatal("Expected error but got nil")
+	} else if len(fixture.updates) != 1 {
+		t.Fatal("expected update")
+	} else if fixture.d.etag != "" {
```

Comment:
nit: I was a bit surprised to see the downloader (which we're testing) on the test _fixture_ struct. Usually fixtures provide the environment required for the system under test (in this case, the downloader is the system under test and the fixture would be the bundle server). It's just odd because the test case is mutating the fixture and it's hard to tell whether it's mutating the downloader at the same time (it's not).

Reply:
If we separate the downloader(system under test) from the test fixture, it would probably be the right thing to do. We can do this as a separate change.